python/cudf/cudf/core/column/column.py

```diff
@@ -1246,8 +1250,10 @@ def corr(self, other):
         )
 
     def nans_to_nulls(self):
-        if self.dtype.kind == "f":
-            col = self.fillna(np.nan)
+        if self.dtype.kind == "f" and libcudf.unary.is_nan(self).any():
+            col = libcudf.replace.replace_nulls(
+                self, cudf.Scalar(self.dtype.type(np.nan))
+            )
             newmask = libcudf.transform.nans_to_nulls(col)
```

Comment:
Maybe I'm misunderstand, but this looks really inefficient. It looks like it first checks if NaNs exist, if they do, replaces nulls with NaN, and then replaces NaNs with nulls?

Reply:
Right, after looking back at the history of the code this is now a redundant operation after `nans_to_nulls` got introduced. Removed this part of the code. Thanks for the catch @jrhemstad 🙏

Reply:
> That seems to be what is happening. Are we able to do this with a single pass instead? For instance, skip the nan count, skip the replace, and just create a new mask for elements which are valid numeric?

I tried generating the mask in a single go by trying `is_valid`, but `np.nan` is also considered as a valid value by that function which is why had to resort to checking and then extracting the mask.


> Also, is `f` the only type we need to check? What about integers and dates/times?

`nan` is only allowed in our float column, so we don't need that check in rest of the numeric types.